src/experimental/information_estimates.cpp

```diff
@@ -0,0 +1,401 @@
+                }
+                else
+                {
+                    tup_vct.push_back(std::make_tuple(idx_free, idx_occ, idx_unk + 1));
+                    acc_prob.push_back(pair.second * un_prop);
+                }
+            }
+        }
+        // Inserting the elements into the map
+        for (int k = 0; k < tup_vct.size(); ++k)
+        {
+            temp_map[tup_vct[k]] = acc_prob[k];
+        }
+    }
+
+    // Leaving in the map only the final outcomes
+    std::unordered_map<map_tuple, kt_double, utils::tuple_hash::HashTuple> out_map;
+    for (auto& pair : temp_map)
+    {
+        int idx_free, idx_occ, idx_unk;
+        std::tie(idx_free, idx_occ, idx_unk) = pair.first;
+        if (idx_free + idx_occ + idx_unk == k)
+        {
+            out_map[pair.first] = pair.second;
+        }
+    }
+    return out_map;
+}
+
+kt_double InformationEstimates::measurementOutcomeEntropy(map_tuple const& meas_outcome)
+{
+    /**
+     * Calculate the measurement outcome entropy
+        * Calculate Log-Odds from initial probability guess
+        * Calculate the probability from those logs
+        * Calculate the entropy with the retrieved probability
+     * Arguments:
+        * meas_outcome [map_tuple]: Measurement outcome in the form {p_free, p_occ, p_unk}
+     * Return:
+        * kt_double: Measurement outcome entropy
+    */
+    int num_free, num_occ, num_unk;
+    std::tie(num_free, num_occ, num_unk) = meas_outcome;
+    kt_double log_occ = (num_free * l_free) + (num_occ * l_occ) - ((num_free + num_occ - 1) * l_o);
+    kt_double prob_occ = calculateProbabilityFromLogOdds(log_occ);
+    return calculateInformationContent(prob_occ);
+}
+
+kt_double InformationEstimates::calculateScanMassProbabilityBetween(kt_double range_1, kt_double range_2)
+{
+    /**
+     * Calculate the mass probability of a cell being observed by a given measurement
+     * Arguments:
+        * range_1 [kt_double]: Lower bound
+        * range_2 [kt_double]: Upper bound
+     * Return:
+        * kt_double: Mass probability
+    */
+    range_1 = (range_1 > m_max_sensor_range) ? m_max_sensor_range : range_1;
+    range_2 = (range_2 > m_max_sensor_range) ? m_max_sensor_range : range_2;
+
+    return m_obs_nu * (exp(-m_obs_lambda*range_1) - exp(-m_obs_lambda*range_2));
+}
+
+std::vector<std::vector<kt_double>> InformationEstimates::retrieveCellProbabilities(karto::Vector2<int> const & cell)
+{
+    /**
+     * Retrieve the cell probabilities (Measurement outcomes)
+     * Arguments:
+        * cell [karto::Vector2<int>]: Cell coordinates
+     * Return:
+        * std::vector<std::vector<kt_double>>: Vector of cell probabilities in the form {p_free, p_occ, p_unk} (Measurement outcomes)
+    */
+    std::map<std::vector<int>, std::vector<std::vector<kt_double>>>::iterator it_cells;
+    it_cells = m_cell_probabilities.find({cell.GetX(), cell.GetY()});
+
+    return it_cells->second;
```

Comment:
@kmilo7204 hmm, isn' this whole thing equivalent to `m_cell_probabilities[{cell.GetX(), cell.GetY()}]` or `m_cell_probabilities.at({cell.GetX(), cell.GetY()})`? Also, why not hash `karto::Vector` instead of creating temporary vectors?